From 7fbf50bf0d8c1dfa3485b2d221f099b43d2a7535 Mon Sep 17 00:00:00 2001
From: Fabien Boucher <fboucher@redhat.com>
Date: Wed, 10 May 2017 17:55:53 +0200
Subject: [PATCH] Set wsme decorator return type to None for some delete calls

This patch fixes Storyboard returning a body along with
a DELETE response 204 No Content for some endpoints.

Co-Authored-By: Matthieu Huin <mhuin@redhat.com>
---
 storyboard/api/v1/stories.py     | 2 +-
 storyboard/api/v1/tasks.py       | 4 ++--
 storyboard/api/v1/timeline.py    | 2 +-
 storyboard/api/v1/user_tokens.py | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/storyboard/api/v1/stories.py b/storyboard/api/v1/stories.py
index de9a53c..3275812 100644
--- a/storyboard/api/v1/stories.py
+++ b/storyboard/api/v1/stories.py
@@ -336,7 +336,7 @@ class StoriesController(rest.RestController):
 
     @decorators.db_exceptions
     @secure(checks.superuser)
-    @wsme_pecan.wsexpose(wmodels.Story, int, status_code=204)
+    @wsme_pecan.wsexpose(None, int, status_code=204)
     def delete(self, story_id):
         """Delete this story. This command is only available to Admin users.
 
diff --git a/storyboard/api/v1/tasks.py b/storyboard/api/v1/tasks.py
index f5ad01e..67d0dc1 100644
--- a/storyboard/api/v1/tasks.py
+++ b/storyboard/api/v1/tasks.py
@@ -417,7 +417,7 @@ class TasksPrimaryController(rest.RestController):
 
     @decorators.db_exceptions
     @secure(checks.authenticated)
-    @wsme_pecan.wsexpose(wmodels.Task, int, status_code=204)
+    @wsme_pecan.wsexpose(None, int, status_code=204)
     def delete(self, task_id):
         """Delete this task.
 
@@ -674,7 +674,7 @@ class TasksNestedController(rest.RestController):
 
     @decorators.db_exceptions
     @secure(checks.authenticated)
-    @wsme_pecan.wsexpose(wmodels.Task, int, int, status_code=204)
+    @wsme_pecan.wsexpose(None, int, int, status_code=204)
     def delete(self, story_id, task_id):
         """Delete this task.
 
diff --git a/storyboard/api/v1/timeline.py b/storyboard/api/v1/timeline.py
index 23fc495..5e91958 100644
--- a/storyboard/api/v1/timeline.py
+++ b/storyboard/api/v1/timeline.py
@@ -406,7 +406,7 @@ class CommentsController(rest.RestController):
 
     @decorators.db_exceptions
     @secure(checks.superuser)
-    @wsme_pecan.wsexpose(wmodels.Comment, int, int, status_code=204)
+    @wsme_pecan.wsexpose(None, int, int, status_code=204)
     def delete(self, story_id, comment_id):
         """Delete an existing comment. This command is disabled by default.
 
diff --git a/storyboard/api/v1/user_tokens.py b/storyboard/api/v1/user_tokens.py
index 4b0632e..1f8830f 100644
--- a/storyboard/api/v1/user_tokens.py
+++ b/storyboard/api/v1/user_tokens.py
@@ -214,7 +214,7 @@ class UserTokensController(rest.RestController):
 
     @decorators.db_exceptions
     @secure(checks.authenticated)
-    @wsme_pecan.wsexpose(wmodels.AccessToken, int, int, status_code=204)
+    @wsme_pecan.wsexpose(None, int, int, status_code=204)
     def delete(self, user_id, access_token_id):
         """Deletes an access token with assigned refresh token for the given
         user. Admin users can delete any access tokens, regular users can only
-- 
2.9.3

